docs/modules/SPEC.md

```diff
@@ -0,0 +1,80 @@
+# Module Specification
+
+This document attempts to outline the recommended structure of Cosmos SDK modules.
+However, the ideas outlined here are meant to be applied as suggestions. Application
+developers are encouraged to improve upon and contribute to module structure and
+development design.
+
+## Structure
+
+A typical Cosmos SDK module can be structured as follows:
+
+```shell
+x/{module}
+├── abci.go
+├── alias.go
+├── client
+│   ├── cli
+│   │   ├── query.go
+│   │   └── tx.go
+│   └── rest
+│       ├── query.go
+│       └── tx.go
+├── exported
+│   └── exported.go
+├── genesis.go
+├── handler.go
+├── internal
```

Comment:
Rather than just sorting all the contents alphabetically I'd recommend also keeping folders at the top of the order in this tree aka. 
```
x/{module}
├── folder a
│   ├── ...
├── folder b
│   ├── ...
├── more folders...
│   ├── ...
├── abci.go
├── alias.go
├── genesis.go
├── handler.go
├── ...

```

Reply:
Sure :)